pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -197,6 +203,26 @@ type QueueingHintFunction struct {
 	QueueingHintFn framework.QueueingHintFn
 }
 
+type inFlightPod struct {
+	// previousEvent is the latest observed event when the pod is popped.
+	previousEvent *list.Element
+}
+
+// clusterEvent has the event and involved objects.
+type clusterEvent struct {
+	requestCycle int64
```

Comment:
add comment explaining that this is the schedulingCycle at the time the event occurred

Reply:
actually, why do we need to track requestCycle at all if we are tracking the events since the pod became inflight? by definition requestCycle of the events happened after the pod was inflight should always be > podSchedulingCycle.

Reply:
Good point.

Actually, the current implementation with requestCycle is following the current implementation. Because requestCycle may not be > podSchedulingCycle in the current impl when the Pod is rejected in the binding cycle.
We have a separated PR for fixing that: https://github.com/kubernetes/kubernetes/pull/119105
I'll remove requestCycle in this PR as well since that simplify this PR as well

Reply:
🙏 https://github.com/kubernetes/kubernetes/pull/118438/commits/6b50795ded92b0620fc97690755ed3a3c1ec4585